In [1]:
import pandas as pd

In [167]:
df = pd.read_csv("D:/SKKU/P-SAT/주제분석/코드 및 데이터/new data/final_with_NE_tagged_quote_count.csv")

In [94]:
quote_count = df['quote_count'].value_counts().sort_index()
quote_count

0     8005
2     4932
4       17
6        3
8        3
10       2
Name: quote_count, dtype: int64

In [49]:
df['easy_sentence'][df['quote_count'] >2 ].to_dict()

{187: '"이 방울이 전에는 그리 덥더니 오늘은 차가움 같아서 아무리 떼이려 하여도 살에 떨림 같아서 떨어지지 않는다”고 하거늘, 무손이 내달아 잡아떼려고 손을 대고자 하니 더욱 불이 성하는 듯하여 손을 대지 못하고 그 처를 잠재우어 말하기를, “가만히 두라. 가만히 두어. 불을 내지 말라.”고 하였다.',
 686: '오늘 새벽에 홍포를 입은 고관 선관이 내려와서 말했어. "남두성이 상제께 죄를 지어서 십 년 동안 허물을 쓰고 세상을 못 보게 했는데, 죄가 너무 심해졌다"고 하고 허물을 벗겨 방 안에 두고 "이 허물을 가져가라. 네 부모에게 보여주어 적절한 사람이 누구인지 알게 하라"고 했어. 그러자 소자 보가 옷을 벗고 보니 허물이 곁에 놓여 있고 책 세 권도 놓여 있었어. 십 년 동안 부모에게 효도하지 못한 죄를 어떻게 다 말해야 할지 모르겠다고 했어.',
 1592: '모든 재상이 이득춘을 향하여 순종하였다. 공은 여러 사람을 이끌어 술을 부어 즐기더니, 해가 지는 곳을 알리자 모든 사람이 각자 집으로 돌아갔다. 이 아들을 데리고 집안으로 들어와 식사를 하고 촛불을 켜 낮을 보내며 즐겼다. 박소저가 외모가 예쁘지 않아서 손을 보기 부끄러워하며 깊이 들어 있음을 서운히 여겨 크게 즐기지 않았다. 부인이 말하기를, "네가 이런 때에 즐기지 않는다니, 이는 무슨 생각이냐?" 하자, "첩의 용모가 부끄러워서입니다." 하였다.',
 1746: '공이 말했어요. "당신은 이 여자를 보고 어떻게 결심하실 건가요?" 한림이 심사숙고한 듯이 말했어요. "진실로 겸손하고 얌전한 여자입니다. 제 손해를 입힐까 두려워할지언정, 어떻게 사양할 수 있겠어요?" 영공이 크게 기뻐하며 환영하고, 한림과 여자를 보러 가기로 했어요. 한림은 아름다우며 여자는 불꽃처럼 빛났어요. 한림은 맑고 깨끗하며 여자는 아름다웠고, 둘은 함께 밝게 빛나는 모습이었어요. 영공이 여자를 보내고 한림을 데리고 종일 즐기며 행복했어요.',
 1747: '어시에 방한림이 중간에 돌아가, 영가 혼란을 주위

SPLIT points
187 하거늘, 말하기를,
686 말했어. 했어. 있었어. 
1592 순종하였다. 돌아갔다. 즐겼다. 않았다. 말하기를, 하자,
1746 말했어요. 말했어요. 있겠어요?" 했어요. 빛났어요. 모습이었어요. 
1747 말했어요. 생각하시나요?" 말했어요, 준비하세요."
1762 없다. 느껴진다. 말했다. 전달했다. 있겠어요."
1767 말했어요. 갖추어야겠네요." 말했어요, 않나요?"
1782 먹었어요. 떠났습니다. 말해요. 말했다. ." 바랐다. 
1786 인물이야. 나이야. 같아. 아름다워. 영웅이야. 말했어. 들어주실래요?" 대답했어. 있겠습니다." 말했어. 허락해주실래요?" 허락했어. 거예요." 부탁해." 바쳤어. 탄복했어. 
1789 말했다. 않나요?" 말했다. 옳은가요?"
1795 말했다, 아시나요?" 대답했다, 
1812 말했다. 하네." 지었다. 않았어." 물었다.
1820 아름다웠어요. 멋있었어요. 말하듯이, 좋을까요?" 말하며, 부족할까요?"
1823 말했어. 이루어라." 말라." 받았어.
1835 말했다. 전하노라." 물러났다. 것이냐?" 한다." 많다." 된다."
1837 보여주었어. 말했어. 허락해주십시오." 대답했어. 
1843 말했다. 계십니까?" 대답했다. 
1845 못했어. 기다렸어. 만났어. 말했어, 거야!" 물었지만, 
2502 말했다. 했다. 
5537 닫아."
8230 물으셨다. 합니다." 사람인가?"
9640 써 주니, 보니,
9930 말했다. 합니다." 같아요." 물었다. 대답했다. 마세요."
11146 생각하되, 어지하리요?" 생각하되, 하더니, 
12474 말했다. 

In [168]:
def split_sentence(row, split_points):
    sentences = [row['easy_sentence']]
    for split_point in split_points:
        new_sentences = []
        for sentence in sentences:
            parts = sentence.split(split_point)
            if len(parts) > 1:
                # split point가 문장 내에 있을 때만 처리
                new_sentences.append(parts[0] + split_point)
                new_sentences.extend(parts[1:])
            else:
                new_sentences.append(parts[0])
        sentences = new_sentences
    new_rows = []
    for sentence in sentences:
        new_rows.append({'title': row['title'], 'easy_sentence': sentence.strip(), 'quote_count':row['quote_count']})
    return new_rows

# split points 설정
split_points = ['하거늘,', '말하기를,']

new_rows = []
for index, row in df.iterrows():
    if index == 187:  
        new_rows.extend(split_sentence(row, split_points))
    else:
        new_rows.append({'title': row['title'], 'easy_sentence': row['easy_sentence'], 'quote_count':row['quote_count']})

df = pd.DataFrame(new_rows)

In [169]:
def split_sentence(row, split_points):
    sentences = [row['easy_sentence']]
    for split_point in split_points:
        new_sentences = []
        for sentence in sentences:
            parts = sentence.split(split_point)
            if len(parts) > 1:
                # split point가 문장 내에 있을 때만 처리
                new_sentences.append(parts[0] + split_point)
                new_sentences.extend(parts[1:])
            else:
                new_sentences.append(parts[0])
        sentences = new_sentences
    new_rows = []
    for sentence in sentences[:-1]:  # 마지막 문장은 그대로 유지
        new_rows.append({'title': row['title'], 'easy_sentence': sentence.strip(), 'quote_count':row['quote_count']})
    new_rows.append({'title': row['title'], 'easy_sentence': sentences[-1].strip(), 'quote_count': row['quote_count']})
    return new_rows

# split points 설정
split_points = ['말했어. ', '했어. ', '있었어. ']

# 새로운 행들을 데이터프레임에 추가
new_rows = []
for index, row in df.iterrows():
    if index == 688:  
        new_rows.extend(split_sentence(row, split_points))
    else:
        new_rows.append({'title': row['title'], 'easy_sentence': row['easy_sentence'], 'quote_count':row['quote_count']})

df = pd.DataFrame(new_rows)

In [170]:
def split_sentence(row, split_points):
    sentences = [row['easy_sentence']]
    for split_point in split_points:
        new_sentences = []
        for sentence in sentences:
            parts = sentence.split(split_point)
            if len(parts) > 1:
                # split point가 문장 내에 있을 때만 처리
                new_sentences.append(parts[0] + split_point)
                new_sentences.extend(parts[1:])
            else:
                new_sentences.append(parts[0])
        sentences = new_sentences
    new_rows = []
    for sentence in sentences:
        new_rows.append({'title': row['title'], 'easy_sentence': sentence.strip(), 'quote_count':row['quote_count']})
    return new_rows

# split points 설정
split_points = ['순종하였다. ', '돌아갔다. ', '즐겼다. ', '않았다. ', '말하기를,', '하자,']

new_rows = []
for index, row in df.iterrows():
    if index == 1598:  
        new_rows.extend(split_sentence(row, split_points))
    else:
        new_rows.append({'title': row['title'], 'easy_sentence': row['easy_sentence'], 'quote_count':row['quote_count']})

df = pd.DataFrame(new_rows)

In [171]:
df['easy_sentence'][df['quote_count'] >2 ].to_dict()

{187: '"이 방울이 전에는 그리 덥더니 오늘은 차가움 같아서 아무리 떼이려 하여도 살에 떨림 같아서 떨어지지 않는다”고 하거늘,',
 188: '무손이 내달아 잡아떼려고 손을 대고자 하니 더욱 불이 성하는 듯하여 손을 대지 못하고 그 처를 잠재우어 말하기를,',
 189: '“가만히 두라. 가만히 두어. 불을 내지 말라.”고 하였다.',
 688: '오늘 새벽에 홍포를 입은 고관 선관이 내려와서 말했어.',
 689: '',
 690: '"남두성이 상제께 죄를 지어서 십 년 동안 허물을 쓰고 세상을 못 보게 했는데, 죄가 너무 심해졌다"고 하고 허물을 벗겨 방 안에 두고 "이 허물을 가져가라. 네 부모에게 보여주어 적절한 사람이 누구인지 알게 하라"고 했어.',
 691: '그러자 소자 보가 옷을 벗고 보니 허물이 곁에 놓여 있고 책 세 권도 놓여 있었어.',
 692: '십 년 동안 부모에게 효도하지 못한 죄를 어떻게 다 말해야 할지 모르겠다고 했어.',
 1598: '모든 재상이 이득춘을 향하여 순종하였다.',
 1599: '공은 여러 사람을 이끌어 술을 부어 즐기더니, 해가 지는 곳을 알리자 모든 사람이 각자 집으로 돌아갔다.',
 1600: '이 아들을 데리고 집안으로 들어와 식사를 하고 촛불을 켜 낮을 보내며 즐겼다.',
 1601: '박소저가 외모가 예쁘지 않아서 손을 보기 부끄러워하며 깊이 들어 있음을 서운히 여겨 크게 즐기지 않았다.',
 1602: '부인이 말하기를,',
 1603: '"네가 이런 때에 즐기지 않는다니, 이는 무슨 생각이냐?" 하자,',
 1604: '"첩의 용모가 부끄러워서입니다." 하였다.',
 1758: '공이 말했어요. "당신은 이 여자를 보고 어떻게 결심하실 건가요?" 한림이 심사숙고한 듯이 말했어요. "진실로 겸손하고 얌전한 여자입니다. 제 손해를 입힐까 두려워할지언정, 어떻게 사양할 수 있겠어요?" 영공이 크게 기뻐하며 환영하고, 한림과 여자를 보러 가기로 했어요. 한림은 아름다우며 여자는 불꽃처럼 빛났어요. 